dex/testing/loadbot/loadbot.go

```diff
@@ -0,0 +1,440 @@
+	var m, n int
+	flag.StringVar(&programName, "p", "", "the bot program to run")
+	flag.BoolVar(&latency500, "latency500", false, "add 500 ms of latency to downstream comms for wallets and server")
+	flag.BoolVar(&shaky500, "shaky500", false, "add 500 ms +/- 500ms of latency to downstream comms for wallets and server")
+	flag.BoolVar(&slow100, "slow100", false, "limit bandwidth on all server and wallet connections to 100 kB/s")
+	flag.BoolVar(&spotty20, "spotty20", false, "drop all connections every 20 seconds")
+	flag.BoolVar(&debug, "debug", false, "use debug logging")
+	flag.BoolVar(&trace, "trace", false, "use trace logging")
+	flag.IntVar(&m, "m", 0, "for compound and sidestacker, m is the number of makers to stack before placing takers")
+	flag.IntVar(&n, "n", 0, "for compound and sidestacker, n is the number of orders to place per epoch (default 3)")
+	flag.Parse()
+
+	if programName == "" {
+		fmt.Println("no program set. use `-p program-name` to set the bot's program")
+		return
+	}
+
+	// Parse the default log level and create initialize logging.
+	logLevel := "info"
+	if debug {
+		logLevel = "debug"
+	}
+	if trace {
+		logLevel = "trace"
+	}
+
+	// Get the server harness configuration file. We'll want these parameters
+	// before we get them from 'config'.
+	f, err := ioutil.ReadFile(filepath.Join(dextestDir, "dcrdex", "markets.json"))
+	if err != nil {
+		fmt.Println("error reading simnet dcrdex markets.json file:", err)
+		return
+	}
+	var mktsCfg *marketsDotJSON
+	err = json.Unmarshal(f, &mktsCfg)
+	if err != nil {
+		fmt.Println("error unmarshaling markets.json:", err)
+		return
+	}
+	for _, mkt := range mktsCfg.Markets {
+		if mkt.Base == "DCR_simnet" && mkt.Quote == "BTC_simnet" {
+			epochDuration = mkt.Duration
+		}
+	}
+	if epochDuration == 0 {
+		fmt.Println("failed to find dcr_btc market in harness config")
+		return
+	}
+
+	btcAssetCfg = mktsCfg.Assets["BTC_simnet"]
+	dcrAssetCfg = mktsCfg.Assets["DCR_simnet"]
+	if dcrAssetCfg == nil || btcAssetCfg == nil {
+		fmt.Println("asset configuration missing from markets.json")
+		return
+	}
+
+	// Load the asset node configs. We'll need the wallet RPC addresses.
+	// Note that the RPC address may be modified if toxiproxy is used below.
+	alphaCfgDCR = loadNodeConfig(dcr, alpha)
+	betaCfgDCR = loadNodeConfig(dcr, beta)
+	alphaCfgBTC = loadNodeConfig(btc, alpha)
+	betaCfgBTC = loadNodeConfig(btc, beta)
+
+	loggerMaker, err = dex.NewLoggerMaker(os.Stdout, logLevel)
+	if err != nil {
+		fmt.Println("error creating LoggerMaker:", err)
+	}
+	log /* global */ = loggerMaker.NewLogger("LOADBOT", dex.LevelInfo)
+
+	log.Infof("Running program %s", programName)
+
+	getAddress := func(symbol, cmd string) string {
+		res := <-harnessCtl(symbol, cmd)
+		if res.err != nil {
+			log.Errorf("error getting address:", symbol, res.err)
+		}
+		return res.output
+	}
+
+	alphaAddrDCR = getAddress(dcr, "./alpha getnewaddress")
+	betaAddrDCR = getAddress(dcr, "./beta getnewaddress")
```

Comment:
Need to add wrap or ignore.  I got an error for a well used dcr harness

`"./alpha", "getnewaddress", "default", "wrap"`